### DISCUSSION TITLE
2.3.21 so-allow remove entry

### DISCUSSION THREAD

**Discussion by sgaitadmins** [Security-Onion-Solutions/securityonion#2625] [Unsupported Versions] [answered]
During testing, I added an entry for Wazuh 1514 and Wazuh 1515 with so-allow for a subnet. I want to take that subnet out now. How do I go about doing this?

**Comment by m0duspwnens** [comment, marked as answer]
Hi @sgaitadmins,
 All of the command below should be run from the manager using sudo.

List the IPs/subnets in the host groups we are wanting to modify and find the IP/subnet you want to remove.

```
so-firewall includedhosts wazuh_agent
so-firewall includedhosts wazuh_authd
```

Remove the IP/subnet from the hostgroup.
```
so-firewall excludehost wazuh_agent IP/subnet
so-firewall excludehost wazuh_authd IP/subnet
```

Since this rule only affects the manager, we can just apply the firewall state to the manager now. If this rule impacted all the nodes int he grid, then we would want to apply the firewall state to all the nodes. 
Apply the firewall state to the manager.
```
salt-call state.apply firewall
```

Double check removal was successful.
```
iptables -nL | grep IP/subnet
```

Since the rule only impacts the manager, and the change has been made by running the firewall state, we can remove the IP/subnet from the excluded hosts list. If the rule impacted more nodes in than just the manager, we would want to leave this in place until out of our nodes had a chance to run the firewall state.
```
so-firewall removehost wazuh_agent IP/subnet
so-firewall removehost wazuh_authd IP/subnet 
```

**Reply by sgaitadmins** [reply]
Exactly what I needed. Answered my question, thank you!
